src/pattern-library/components/Library.tsx

```diff
@@ -332,48 +410,95 @@ function ChangelogItem({ status, children }: LibraryChangelogItemProps) {
   );
 }
 
+type CodeContentProps =
+  | {
+      /** Code content (to be rendered with syntax highlighting) */
+      content: ComponentChildren;
+    }
+  | {
+      /**
+       * Example file to read and use as content (to be rendered with syntax
+       * highlighting)
+       */
+      exampleFile: string;
+    };
+
 export type LibraryCodeProps = {
-  /** Code content (to be rendered with syntax highlighting) */
-  content: ComponentChildren;
   /** Controls relative code font size */
   size?: 'sm' | 'md';
   /** Caption (e.g. filename, description) of code block */
   title?: ComponentChildren;
-};
+} & CodeContentProps;
+
+function isCodeWithContent(
+  props: CodeContentProps,
+): props is { content: ComponentChildren } {
+  return 'content' in props;
+}
+
+/**
+ * Dynamically resolves the content based on provided props.
+ * An error is optionally returned in case loading the content failed.
+ */
+function useCodeContent(
+  props: CodeContentProps,
+): [string | undefined, Error | undefined] {
```

Comment:
This function could return `undefined | string | Error` instead, to reflect the three possible states (loading, successful, failed).

Reply:
It started like that, but I ended up moving to this approach for some reason I don't remember now 🤔 😅 

Let me check again.

Reply:
Oh, yeah. I moved to this approach because otherwise you end up with a `codeMarkupOrError` var, which needs to be type-checked in order to decide what to render.

Right now it looks like this:

```ts
const [codeMarkup, error] = useCodeContent(rest);

return (
  {codeMarkup && (...)}
  {error && <SimpleError message={error.message} />}
);
```

Using just one var, it would look like this:

```ts
const codeMarkupOrError = useCodeContent(rest);

return (
  {typeof codeMarkupOrError === 'string' && (...)}
  {typeof codeMarkupOrError === 'object' && <SimpleError message={codeMarkupOrError.message} />}
);
```

Which is a bit harder to read. I'll leave it as is, but we can change it afterwards if needed.